templates/pkg/resource/sdk_update.go.tpl

```diff
@@ -5,6 +5,9 @@ func (rm *resourceManager) sdkUpdate(
 	latest *resource,
 	diffReporter *ackcompare.Reporter,
 ) (*resource, error) {
+{{- if $hookCode := .CRD.HookCode sdk_update_pre_build_request }}
+{{ $hookCode }}
+{{ end -}}
```

Comment:
Obviously I've only added in a single `sdk_update_pre_build_request` hook here, into the `sdk_update.go.tpl` template. If folks this is a good strategy, we can go and add other hook points in the other templates.

Reply:
I believe we support x-ray distributed tracing, correct? If so, would we propagate the tracing to the hook?

Reply:
> I believe we support x-ray distributed tracing, correct? If so, would we propagate the tracing to the hook?

I think we only have Prometheus metrics for now. Regarding tracing I have two questions:
- what if containers are running outside of AWS? It is still possible to propagate tracing?
- if yes, what if customers want to use a different solution of tracing (jager ....etc)?

Reply:
> I believe we support x-ray distributed tracing, correct? If so, would we propagate the tracing to the hook?

@panazzo this is for injecting custom code into known places in the various templates. If we wanted to support x-ray tracing, we'd probably want to do that in a standardized way -- similar to how we currently have a Prometheus standard way of tracking API calls:

https://github.com/aws-controllers-k8s/runtime/blob/cbda0dfbbca384a56c863b3973d8e3c437a46088/pkg/metrics/metrics.go#L63-L90

Happy to have a conversation about x-ray support in a separate github issue!